tests/src/test/scala/org/apache/spark/sql/rapids/metrics/source/MockTaskContext.scala

```diff
@@ -29,7 +29,7 @@ import org.apache.spark.scheduler.TaskLocality
 import org.apache.spark.shuffle.FetchFailedException
 import org.apache.spark.util.{AccumulatorV2, TaskCompletionListener, TaskFailureListener}
 
-class MockTaskContext(taskAttemptId: Long, partitionId: Int) extends TaskContext {
```

Comment:
When there is just a new method that has not been used previously, we do not really have to fork the implementation into two shims as long as we do not insist on using the override keyword.

```Scala
class MockTaskContext(taskAttemptId: Long, partitionId: Int) extends TaskContext {
   def isFailed(): Boolean = false
}
```